lib/kube/proxy/auth_test.go

```diff
@@ -125,15 +130,57 @@ func failsForCluster(clusterName string) ImpersonationPermissionsChecker {
 	}
 }
 
+func createFileWithContents(t *testing.T, contents []byte) string {
+	tmpFile, err := os.CreateTemp("", "teleport")
+	require.NoError(t, err)
+	t.Cleanup(func() {
+		os.Remove(tmpFile.Name())
+	})
+	_, err = tmpFile.Write(contents)
```

Comment:
We need an explicit close after the Write, plus we also need to make sure the file closes in other scenarios. 

I suggest you use `os.WriteFile`, it sidesteps all of those.